contracts/SimpleOperations.sol

```diff
@@ -45,121 +46,258 @@ contract SimpleOperations is Operations {
-		return latestInTrack(_client, track(_client, _release)) == _release;
+	function isLatest(bytes32 _client, bytes32 _release)
+		public
+		view
+		returns (bool)
+	{
+		return clientExists(_client) && latestInTrack(_client, track(_client, _release)) == _release;
 	}
 
-	function track(bytes32 _client, bytes32 _release) public view returns (uint8) {
-		return client[_client].release[_release].track;
+	function track(bytes32 _client, bytes32 _release)
+		public
+		view
+		returns (uint8)
+	{
+		if (clientExists(_client)) {
+			return client[_client].release[_release].track;
+		} else {
+			return 0;
+		}
 	}
 
-	function latestInTrack(bytes32 _client, uint8 _track) public view returns (bytes32) {
-		return client[_client].current[_track];
+	function latestInTrack(bytes32 _client, uint8 _track)
+		public
+		view
+		returns (bytes32)
+	{
+		if (clientExists(_client)) {
+			return client[_client].current[_track];
+		} else {
+			return 0;
+		}
 	}
 
-	function build(bytes32 _client, bytes32 _checksum) public view returns (bytes32 o_release, bytes32 o_platform) {
-		var b = client[_client].build[_checksum];
-		o_release = b.release;
-		o_platform = b.platform;
+	function build(bytes32 _client, bytes32 _checksum)
+		public
+		view
+		returns (bytes32 o_release, bytes32 o_platform)
+	{
+		if (clientExists(_client)) {
+			Build storage b = client[_client].build[_checksum];
+			o_release = b.release;
+			o_platform = b.platform;
+		} else {
+			o_release = bytes32(0);
+			o_platform = bytes32(0);
+		}
 	}
 
-	function release(bytes32 _client, bytes32 _release) public view returns (uint32 o_forkBlock, uint8 o_track, uint24 o_semver, bool o_critical) {
-		var b = client[_client].release[_release];
-		o_forkBlock = b.forkBlock;
-		o_track = b.track;
-		o_semver = b.semver;
-		o_critical = b.critical;
+	function release(bytes32 _client, bytes32 _release)
+		public
+		view
+		returns (
+			uint32 o_forkBlock,
+			uint8 o_track,
+			uint24 o_semver,
+			bool o_critical
+		)
+	{
+		if (clientExists(_client)) {
+			Release storage r = client[_client].release[_release];
+			o_forkBlock = r.forkBlock;
+			o_track = r.track;
+			o_semver = r.semver;
+			o_critical = r.critical;
+		} else {
+			o_forkBlock = 0;
+			o_track = 0;
+			o_semver = 0;
+			o_critical = false;
+		}
 	}
 
-	function checksum(bytes32 _client, bytes32 _release, bytes32 _platform) public view returns (bytes32) {
-		return client[_client].release[_release].checksum[_platform];
+	function checksum(bytes32 _client, bytes32 _release, bytes32 _platform)
+		public
+		view
+		returns (bytes32)
+	{
+		if (clientExists(_client)) {
+			return client[_client].release[_release].checksum[_platform];
+		} else {
+			return bytes32(0);
+		}
 	}
 
-	function latestFork() public view returns (uint32) {
+	function latestFork()
+		public
+		view
+		returns (uint32)
+	{
 		return latestFork;
 	}
 
-	function clientOwner(address _owner) public view returns (bytes32) {
+	function clientOwner(address _owner)
```

Comment:
Function name is a bit misleading imho, `clientExists` takes `_client` and returns true/false, `clientOwner` however takes `_owner` and returns a client name. Why do we need a function for this anyway?

Reply:
I agree that the function name is misleading, it's a carryover from the naming of the state variables. IMO `client` and `clientOwner` should have their names switched with one another. This method is defined in the interface, and the `OperationsProxy` uses it to get the name of the client it is managing.